src/lib/config/root-config.ts

```diff
@@ -26,6 +44,31 @@ export class RootConfig {
       );
     }
   }
+
+  /**
+   * If there is an endpoint object in the config, and it has a
+   * object underneath it, we are saying that it uses the
+   *
+   * [endpoint.myacct]
+   * secret=***
+   *
+   * structure. This allows us to support a legacy config that has
+   * a top level endpoint named endpoint.  Once we rewrite the config
+   * to be endpoint object based adding one manually like that
+   * will not be recognized.
+   */
+  private static configContainsNestedEndpointStructure(
+    config: Config
+  ): boolean {
+    if (config.objectExists("endpoint")) {
+      const endpointObj = config.object("endpoint");
+      return endpointObj.keys().some((key) => {
+        return endpointObj.objectExists(key);
+      });
```

Comment:
style (you can avoid the second `return` if you don't make a block):
```suggestion
      return endpointObj.keys().some((key) => endpointObj.objectExists(key));
```